CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -248,6 +248,10 @@ protected static function clearGroupContactCache($groupID): void {
    * clear.
    */
   protected static function flushCaches() {
+    if (!CRM_Core_Config::isPermitCacheFlushMode()) {
+      return;
+    }
```

Comment:
@eileenmcnaughton Do you think this bit is ok as-is? If so I'll split it off into it's own PR and work through the more complex bits separately.

Reply:
@mattwire yeah I think it is

Reply:
ok see https://github.com/civicrm/civicrm-core/pull/21468